### ISSUE TITLE
[AI] Profiler runtime events fall back to a generic "cudaRuntime" name (180/203) where torch+CUDA names every call

### ISSUE THREAD

**Issue by lvyufeng** [flagos-ai/Torch-FL#186]
## Issue Type
- [x] Bug Report
- [ ] Feature Request
- [ ] Operator Implementation
- [ ] Platform Support
- [ ] Performance Issue
- [ ] Documentation

## AI Agent Information
- **Agent**: Claude Code CLI
- **Model**: Claude Opus 5 (1M context)
- **Session Context**: While fixing #182 (see #185) the maintainer asked that the flagos
  profiler match stock torch+CUDA behavior. A differential run of one identical workload
  on both stacks confirmed memset parity is exact, but surfaced this separate gap.

## Summary
89% of `privateuse1_runtime` events (180 of 203 in a plain matmul workload) are exported
with the generic name `cudaRuntime` instead of the actual CUDA runtime API name. Stock
torch+CUDA names every one of the equivalent `cuda_runtime` events. The cause is that
`cuptiRuntimeCbidToName()` in `csrc/profiler/cupti_shim.h` is a hand-written table of
about 20 callback ids, and every unmapped id falls through to `default: return
"cudaRuntime"`. All 10 unmapped ids seen in this workload correspond to real, ordinary
entries in CUPTI's official `cupti_runtime_cbid.h`. This degrades trace readability in
Chrome trace / Perfetto and in `key_averages()` output, where distinct API calls collapse
into one indistinguishable label.

## Environment
<details>
<summary>Click to expand environment details</summary>

- **Platform**: CUDA
- **Python**: 3.12
- **PyTorch**: 2.10.0+cpu (with external cu128 `libtorch_cuda.so`)
- **torch_fl**: `4a71161` (main)
- **Hardware**: NVIDIA A100-SXM4-40GB

**Runtime config:**
```bash
FLAGOS_BACKEND_CONFIG=torch_fl/configs/backends_cuda.conf
```
</details>

## Reproduction
```python
import json, tempfile
from pathlib import Path
from collections import Counter
import torch_fl, torch

dev = torch.device("flagos", 0)
x = torch.randn(1024, 1024, device=dev)
y = torch.randn(1024, 1024, device=dev)
small = torch.randn(16, device=dev)

with torch.profiler.profile(
    activities=[torch.profiler.ProfilerActivity.CPU,
                torch.profiler.ProfilerActivity.PrivateUse1],
    with_stack=False) as prof:
    for _ in range(3):
        z = (x @ y).relu()
    torch.sort(small)
    z.sum().item()

with tempfile.NamedTemporaryFile(suffix=".json", delete=False) as f:
    p = Path(f.name)
prof.export_chrome_trace(str(p))
trace = json.loads(p.read_text())
p.unlink(missing_ok=True)

rt = [e for e in trace["traceEvents"]
      if e.get("cat") == "privateuse1_runtime" and e.get("ph") == "X"]
generic = [e for e in rt if e.get("name") == "cudaRuntime"]
print("total runtime:", len(rt), " generic:", len(generic))
print(Counter((e.get("args") or {}).get("cbid") for e in generic).most_common())
```

## Observed
```
total runtime: 203  generic: 180
[(17, 94), (200, 35), (134, 18), (10, 16), (251, 8), (210, 4), (406, 2), (53, 1), (317, 1), (15, 1)]
```

Resolving those ids against `cupti_runtime_cbid.h` from `nvidia-cuda-cupti-cu12`:

| cbid | count | CUPTI symbol |
|-----:|------:|--------------|
| 17 | 94 | `cudaGetDevice_v3020` |
| 200 | 35 | `cudaDeviceGetAttribute_v5000` |
| 134 | 18 | `cudaEventCreateWithFlags_v3020` |
| 10 | 16 | `cudaGetLastError_v3020` |
| 251 | 8 | `cudaOccupancyMaxActiveBlocksPerMultiprocessorWithFlags_v7000` |
| 210 | 4 | `cudaOccupancyMaxActiveBlocksPerMultiprocessor_v6050` |
| 406 | 2 | `cudaGetDriverEntryPoint_v11030` |
| 53 | 1 | `cudaGetSymbolAddress_v3020` |
| 317 | 1 | `cudaStreamIsCapturing_v10000` |
| 15 | 1 | `cudaFuncGetAttributes_v3020` |

## Expected
Stock torch+CUDA, same workload, names every runtime event. Per-name diff of the two
traces (entries present in one stack's naming but not the other):

```
cudaDeviceGetAttribute                          cuda=25  flagos=0
cudaFuncGetAttributes                           cuda= 1  flagos=0
cudaGetDriverEntryPoint                         cuda= 2  flagos=0
cudaGetSymbolAddress                            cuda= 1  flagos=0
cudaHostAlloc                                   cuda= 1  flagos=0
cudaOccupancyMaxActiveBlocksPerMultiprocessor   cuda= 4  flagos=0
cudaOccupancyMaxActiveBlocksPerMultiprocessorWithFlags  cuda=8  flagos=0
cudaStreamIsCapturing                           cuda= 1  flagos=0
cudaStreamSynchronize                           cuda= 1  flagos=0
cudaRuntime                                     cuda= 0  flagos=180
```

Note the `args` keys already match upstream exactly (`External id`, `cbid`,
`correlation`), and the raw numeric `cbid` is preserved on every event — so the
information needed to name these is present in the trace; only the lookup is missing.

## Why the existing test does not catch it
`test_profiler_runtime_names_are_not_all_fallback` asserts
`len(names) > 1 or "cbid" in arg_key_union(...)`. Both disjuncts hold at 180/203
degraded, so the test passes. The assertion was written to catch a total-collapse
regression, not partial degradation.

## Suggested fix
1. Generate the cbid -> name table from CUPTI's `cupti_runtime_cbid.h` (486 entries,
   410 distinct API names after stripping `_vNNNNN` / `_ptsz` suffixes) instead of
   maintaining it by hand. This removes the whole class of bug rather than the 10
   instances observed here, and the header is the authoritative source, so it also
   removes the "a plausible but wrong label is emitted rather than an error" hazard
   the current code comments warn about.
2. Tighten the test to bound the degraded fraction (e.g. `cudaRuntime` must be under
   some small percentage of runtime events) so partial regressions are caught.

Scope notes: this affects the NVIDIA table only. MetaX has its own id space and
normally resolves names through `mcptiActivityGetApiName`, with
`mcptiRuntimeCbidToName` as a legacy fallback; that path should be left alone.

## Related
- Related to #182 and #185 (memset workload sizing — same investigation, separate cause)
- Deliberately excluded from #185 to keep that fix reviewable: this one touches C++ and
  changes an assertion, so it needs a rebuild and its own verification.

## Two other divergences found in the same diff (recorded here, not proposed for fix)
1. **Flow arrow pairing**: stock CUDA emits 19 `s` / 65 `f` (unpaired); flagos emits
   17/17 (paired). flagos is *stricter* here, not lossy — `docs/profiler.md` documents
   this as an intentional flagos-specific invariant.
2. **`overhead` category absent**: known and already justified in
   `tests/data/profiler_cuda_baseline.json` — `CUPTI_ACTIVITY_KIND_OVERHEAD` is never
   enabled, and those records are CUPTI's own profiling overhead, not user work.

Also observed: flagos records roughly 2x the `cpu_op` events of stock CUDA
(`aten::mm` 6 vs 3) because PrivateUse1 boxing re-enters the dispatcher, recording the
same op at two layers. That is architectural rather than a tracer defect, but if strict
upstream parity is a goal it deserves its own evaluation.
